os/fs/smartfs/smartfs_utils.c

```diff
@@ -752,7 +752,12 @@ int smartfs_sync_internal(struct smartfs_mountpt_s *fs, struct smartfs_ofile_s *
 			header->used[1] = (uint8_t)(tmp >> 8);
 		}
 #endif
-		smartfs_setbuffer(&readwrite, sf->currsector, offsetof(struct smartfs_chain_header_s, used), sizeof(uint16_t), (uint8_t *)&fs->fs_rwbuffer[offsetof(struct smartfs_chain_header_s, used)]);
+		if (header->type == CONFIG_SMARTFS_ERASEDSTATE) {
```

Comment:
anyway dynamic header also need to be removed also if it enabled, we can' use new code.
so your code can be merged to #else part.

Reply:
Resolved, thankyou